src/git-town-append

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+source "$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/helpers/helpers.sh"
+
+
+function ensure_has_target_branch {
+  if [ -z "$target_branch_name" ]; then
+    echo_error_header
+    echo_error "No branch name provided."
+    exit_with_error newline
+  fi
+}
+
+
+function preconditions {
+  target_branch_name=$1
+  ensure_has_target_branch
+
+  parent_branch_name=$INITIAL_BRANCH_NAME
+  if [ "$parent_branch_name" != "$MAIN_BRANCH_NAME" ]; then
+    ensure_knows_parent_branches "$parent_branch_name"
+  fi
+
+  if [ "$(has_remote_url)" = true ]; then
+    fetch
+  fi
+
+  ensure_does_not_have_branch "$target_branch_name"
+
+  export RUN_IN_GIT_ROOT=true
+  export STASH_OPEN_CHANGES=true
+}
+
+
+function steps {
+
+  # sync the current branch
```

Comment:
Okay. I think we need a new function to encapslate this in another function then which syncs the branch passed in and its ancestors. That can be dealt with separately,

This comment is also confusing to me as it says "current branch" but we the variable being used is "parent branch". At this point we have saved the current branch into a parent branch variable but thats not obvious to me. Can we just use INITIAL_BRANCH_NAME everywhere instead of assigning the value to parent branch?